kernel/sched.c

```diff
@@ -563,8 +563,9 @@ void _priq_dumb_remove(sys_dlist_t *pq, struct k_thread *thread)
 
 struct k_thread *_priq_dumb_best(sys_dlist_t *pq)
 {
-	return CONTAINER_OF(sys_dlist_peek_head(pq),
-			    struct k_thread, base.qnode_dlist);
+	sys_dnode_t *n = sys_dlist_peek_head(pq);
+
+	return CONTAINER_OF(n, struct k_thread, base.qnode_dlist);
```

Comment:
Same pattern again.  What is is that is forcing you to make the first argument to CONTAINER_OF be a local variable?  We have to fix that some other way.